Launcher doesn't process a long list of command line args properly

### Summary
When passed a long list of command line arguments, the launcher processes the first 8 or 9 arguments properly but provides incorrect values to the launched process for subsequent arguments. 

### Reproduction

##### Steps
This issue can be reproduced by modifying the `launcher_test.go` [acceptance tests case that validates passed args](https://github.com/buildpacks/lifecycle/blob/main/acceptance/launcher_test.go#L55-L65) to look like this: 

```
			it("appends any args to the process args", func() {
				cmd := exec.Command("docker", "run", "--rm",
					"--entrypoint=web",
					"--env=CNB_PLATFORM_API=0.4",
					launchImage,
					"--arg1",
					"--arg2",
					"--arg3",
					"--arg4",
					"--arg5",
					"--arg6",
					"--arg7",
					"--arg8",
					"--arg9",
					"--arg10")
				if runtime.GOOS == "windows" {
					assertOutput(t, cmd, `Executing web process-type "--arg1" "--arg2" "--arg3" "--arg4" "--arg5" "--arg6" "--arg7" "--arg8" "--arg9" "--arg10"`)
				} else {
					assertOutput(t, cmd, "Executing web process-type --arg1 --arg2 --arg3 --arg4 --arg5 --arg6 --arg7 --arg8 --arg9 --arg10")
				}
			})
```

**Note**: I've only tested this with `runtime.GOOS == "linux"`. 

##### Current behavior
```
        --- FAIL: TestLauncher/acceptance/Platform_API_>=_0.4/entrypoint_is_a_process/appends_any_args_to_the_process_args (2.82s)
            launcher_test.go:73: failed:
                	 output: sourced bp profile
                sourced app profile
                Executing web process-type --arg1 --arg2 --arg3 --arg4 --arg5 --arg6 --arg7 --arg8 Executing web process-type0 Executing web process-type1
                
                	 should include: Executing web process-type --arg1 --arg2 --arg3 --arg4 --arg5 --arg6 --arg7 --arg8 --arg9 --arg10
```

This failed test result is consistent with what is seen when running an image built with a builder - the first 8 or 9 args are passed properly but after that the launched process receives the value of first arg with an ordinal (`0`, `1`, etc) appended to it. 

##### Expected
All command-line arguments should be passed to the launched process.

--- a/launch/bash.go
+++ b/launch/bash.go
@@ -53,7 +53,7 @@ func (b *BashShell) Launch(proc ShellProcess) error {
 func bashCommandWithTokens(nTokens int) string {
 	commandScript := `"$(eval echo \"$0\")"`
 	for i := 1; i < nTokens; i++ {
-		commandScript += fmt.Sprintf(` "$(eval echo \"$%d\")"`, i)
+		commandScript += fmt.Sprintf(` "$(eval echo \"${%d}\")"`, i)
 	}
 	return fmt.Sprintf(`exec bash -c '%s' "${@:1}"`, commandScript)
 }

--- a/launch/bash_test.go
+++ b/launch/bash_test.go
@@ -130,6 +130,37 @@ func testBash(t *testing.T, when spec.G, it spec.S) {
 				}
 				h.AssertStringContains(t, stdout, "SOME_ARG: 'some arg1'")
 			})
+
+			it("handles many args", func() {
+				process = launch.ShellProcess{
+					Script:  false,
+					Command: `echo`,
+					Args: []string{
+						"one",
+						"two",
+						"three",
+						"four",
+						"five",
+						"six",
+						"seven",
+						"eight",
+						"nine",
+						"ten",
+					},
+					Caller: "some-profile-argv0",
+					Env: []string{
+						"SOME_VAR=some-val",
+					},
+				}
+				err := shell.Launch(process)
+				h.AssertNil(t, err)
+				stdout := rdfile(t, filepath.Join(tmpDir, "stdout"))
+				if len(stdout) == 0 {
+					stderr := rdfile(t, filepath.Join(tmpDir, "stderr"))
+					t.Fatalf("stdout was empty: stderr: %s\n", stderr)
+				}
+				h.AssertStringContains(t, stdout, "one two three four five six seven eight nine ten")
+			})
 		})
 
 		when("is not script", func() {

